cmd/zpool/zpool_main.c

```diff
@@ -2479,61 +2498,595 @@ zpool_do_import(int argc, char **argv)
+	}
+	free(newnva);
+	free(oldnva);
+
+	return (calcnva);
+}
+
+static void
+free_calc_stats(struct stat_array *nva, unsigned int len)
+{
+	int i;
+	for (i = 0; i < len; i++)
+		free(nva[i].data);
 
-	zfs_nicenum(value, buf, sizeof (buf));
-	(void) printf("  %5s", buf);
+	free(nva);
+}
+
+static void
+print_iostat_histo(struct stat_array *nva, unsigned int len,
+    iostat_cbdata_t *cb, unsigned int column_width, unsigned int namewidth,
+    double scale)
+{
+	int i, j;
+	char buf[6];
+	uint64_t val;
+	enum zfs_nicenum_format format;
+	unsigned int buckets;
+
+	if (cb->cb_literal)
+		format = ZFS_NICENUM_RAW;
+	else
+		format = ZFS_NICENUM_1024;
+
+	/* All these histos are the same size, so just use nva[0].len */
+	buckets = nva[0].len;
+
+	for (j = 0; j < buckets; j++) {
+		/* Ending range of this bucket */
+		val = (1UL << (j + 1)) - 1;
+
+		/* Print histogram bucket label */
+		zfs_nicetime(val, buf, sizeof (buf));
+		if (cb->cb_scripted)
+			printf("%llu", (u_longlong_t) val);
+		else
+			printf("%-*s", namewidth, buf);
+
+		/* Print the values on the line */
+		for (i = 0; i < len; i++) {
+			print_one_stat(nva[i].data[j] * scale, format,
+			    column_width, cb->cb_scripted);
+		}
+		printf("\n");
+	}
+}
+
+static void
+print_solid_separator(unsigned int length)
+{
+	while (length--)
+		printf("-");
+	printf("\n");
+}
+
+static void
+print_iostat_histos(iostat_cbdata_t *cb, nvlist_t *oldnv,
+    nvlist_t *newnv, double scale, const char *name)
+{
+	unsigned int column_width;
+	unsigned int namewidth;
+	unsigned int entire_width;
+
+	const char *names[] = {
+		ZPOOL_CONFIG_TOT_R_LAT_HISTO,
+		ZPOOL_CONFIG_TOT_W_LAT_HISTO,
+		ZPOOL_CONFIG_DISK_R_LAT_HISTO,
+		ZPOOL_CONFIG_DISK_W_LAT_HISTO,
+		ZPOOL_CONFIG_SYNC_R_LAT_HISTO,
+		ZPOOL_CONFIG_SYNC_W_LAT_HISTO,
+		ZPOOL_CONFIG_ASYNC_R_LAT_HISTO,
+		ZPOOL_CONFIG_ASYNC_W_LAT_HISTO,
+		ZPOOL_CONFIG_SCRUB_LAT_HISTO,
+	};
+	struct stat_array *nva;
+	nva = calc_and_alloc_stats(names, ARRAY_SIZE(names), oldnv, newnv);
+
+	if (cb->cb_literal) {
+		column_width = MAX(5,
+		    (unsigned int) log10(stat_histo_max(nva,
+		    ARRAY_SIZE(names))) + 1);
+	} else {
+		column_width = 5;
+	}
+
+	namewidth = MAX(cb->cb_namewidth, strlen("latency"));
+
+	/*
+	 * Calculate the entire line width of what we're printing.  The
+	 * +2 is for the two spaces between columns:
+	 */
+	/*	 read  write				*/
+	/*	-----  -----				*/
+	/*	|___|  <---------- column_width		*/
+	/*						*/
+	/*	|__________|  <--- entire_width		*/
+	/*						*/
+	entire_width = namewidth + (column_width + 2) *
+	    str_array_len(iostat_bottom_labels[IOS_L_HISTO]);
+
+	if (cb->cb_scripted)
+		printf("%s\n", name);
+	else
+		print_iostat_header_impl(cb, column_width, name);
+
+	print_iostat_histo(nva, ARRAY_SIZE(names), cb, column_width,
+	    namewidth, scale);
+
+	free_calc_stats(nva, ARRAY_SIZE(names));
+	if (!cb->cb_scripted)
+		print_solid_separator(entire_width);
+}
+
+/* Calculate the average latency of a power-of-two latency histogram */
```

Comment:
[style] For function prototype block comments use the multi-line format even if it's just a single line.  This doesn't apply to comments inside functions.

```
/*
 * Calcualte...
 */
```